### PR TITLE
Feature: Check if bucket exists with S3Template

### PR DESCRIPTION
**Pull request by straurob**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Added a method to the `S3Template` which allows to check if a bucket exists by a given bucket name. The method returns a simple `boolean`.

## :bulb: Motivation and Context
Please refer to #901 

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Operations.java [hunk 1]
```diff
@@ -47,6 +47,14 @@ public interface S3Operations {
 	 */
 	void deleteBucket(String bucketName);
 
+	/**
+	 * Checks if an S3 bucket exists.
+	 *
+	 * @param bucketName - the bucket name
+	 * @return true if bucket exists; false otherwise
+	 */
+	boolean existsBucket(String bucketName);
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Operations.java:56]: `existsBucket` does not sound right. Perhaps `bucketExists` would be better? @MatejNedic opinion welcome

Comment (straurob) [reply spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Operations.java:56]: I agree that bucketExists is a much more readable and comprehensive name. Will change it next weekend 🙂

Comment (straurob) [reply spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Operations.java]: Updated the method name to `bucketExists`

**Review by maciejwalkowiak — APPROVED** [review]
Thank you @straurob!

**Review by MatejNedic — APPROVED** [review]

**Pull request merged**